src/Illuminate/Support/Str.php

```diff
@@ -841,25 +841,37 @@ public static function pluralStudly($value, $count = 2)
      */
     public static function password($length = 32, $letters = true, $numbers = true, $symbols = true, $spaces = false)
     {
-        return (new Collection)
-                ->when($letters, fn ($c) => $c->merge([
-                    'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k',
-                    'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v',
-                    'w', 'x', 'y', 'z', 'A', 'B', 'C', 'D', 'E', 'F', 'G',
-                    'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R',
-                    'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z',
-                ]))
-                ->when($numbers, fn ($c) => $c->merge([
-                    '0', '1', '2', '3', '4', '5', '6', '7', '8', '9',
-                ]))
-                ->when($symbols, fn ($c) => $c->merge([
-                    '~', '!', '#', '$', '%', '^', '&', '*', '(', ')', '-',
-                    '_', '.', ',', '<', '>', '?', '/', '\\', '{', '}', '[',
-                    ']', '|', ':', ';',
-                ]))
-                ->when($spaces, fn ($c) => $c->merge([' ']))
-                ->pipe(fn ($c) => Collection::times($length, fn () => $c[random_int(0, $c->count() - 1)]))
-                ->implode('');
+        $password = new Collection();
+
+        $options = (new Collection([
+            'letters' => $letters === true ? [
+                'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k',
+                'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v',
+                'w', 'x', 'y', 'z', 'A', 'B', 'C', 'D', 'E', 'F', 'G',
+                'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R',
+                'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z',
+            ] : null,
+            'numbers' => $numbers === true ? [
+                '0', '1', '2', '3', '4', '5', '6', '7', '8', '9',
+            ] : null,
+            'symbols' => $symbols === true ? [
+                '~', '!', '#', '$', '%', '^', '&', '*', '(', ')', '-',
+                '_', '.', ',', '<', '>', '?', '/', '\\', '{', '}', '[',
+                ']', '|', ':', ';',
+            ] : null,
+            'spaces' => $spaces === true ? [' '] : null,
```

Comment:
It seems to be sufficient to determine if a value is true or not:

```suggestion
        $options = (new Collection([
            'letters' => $letters ? [
                'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k',
                'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v',
                'w', 'x', 'y', 'z', 'A', 'B', 'C', 'D', 'E', 'F', 'G',
                'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R',
                'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z',
            ] : null,
            'numbers' => $numbers ? [
                '0', '1', '2', '3', '4', '5', '6', '7', '8', '9',
            ] : null,
            'symbols' => $symbols ? [
                '~', '!', '#', '$', '%', '^', '&', '*', '(', ')', '-',
                '_', '.', ',', '<', '>', '?', '/', '\\', '{', '}', '[',
                ']', '|', ':', ';',
            ] : null,
            'spaces' => $spaces ? [' '] : null,
```

Since there is no native PHP type declaration, it is safer to treat all truthy values as true.

Reply:
but the parameter isn't typehinted as `bool`

Reply:
> but the parameter isn't typehinted as `bool`

Yes, I rather think so, which is **why `=== true` should be omitted**. For example, if you pass `(int)1` here, I think it should be handled as `true` because it is a truthy value.

In a situation where either `true` or `false` is determined, it seems somewhat unnatural that all truthy values other than `true` would be `false`. It would be consistent in the case of adopting a native PHP type constraint, where `1` would be cast to `true` at the start of function execution.

Reply:
https://github.com/laravel/framework/blob/09137f50f715c1efc649788a26092dcb1ec4ab6e/src/Illuminate/Support/Str.php#L842

What's your expectation with without `=== true` and the following example:

```php
Str::password(numbers: 'nah');
```

Reply:
`Str::password(numbers: 'nah');` should be `Str::password(numbers: true);` because PHP treats '"nah"' as truthy. The important thing is that **we should follow the interpretation of the PHP standard, not determine "what should be true" at the the framework.** You would be correct if Laravel were to adopt the rule of always treating truthy values other than `true` as `false`, but it is not actually happened so far.

Reply:
> You would be correct if Laravel were to adopt the rule of always treating truthy values other than true as false, but it is not actually happened so far

I would disagree https://github.com/search?q=repo%3Alaravel%2Fframework%20%3D%3D%3D%20true&type=code

Reply:
@crynobone I see, that is certainly the case in some codes. But apparently it is not consistent across the board. Somewhat confused.

https://github.com/search?q=repo%3Alaravel%2Fframework+path%3A*.php+%2Fif+%5C%28%5C%24%5Cw%2B%5C%29%2F&type=code

For the moment, I see that there seems to be no need to discuss this excessively here. I think this has been an off-topic discussion, but thank you for your patience.